src/Content/LabPage.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\Collection;
+use Kirby\Cms\Helpers;
+use Kirby\Cms\Language;
+use Kirby\Cms\Page;
+
+/**
+ * Test Model to prototype the content and translation
+ * mechanics
+ */
+class LabPage extends Page
+{
+	/**
+	 * Creates a new instance with the same
+	 * initial properties
+	 *
+	 * @todo eventually refactor without need of propertyData
+	 */
+	public function clone(array $props = []): static
+	{
+		// keep the current state of the content in memory storage
+		// before creating the clone. The storage of the clone can
+		// afterwards be mutated without affecting the content in the
+		// original instance.
+		$this->storage = MemoryContentStorageHandler::from($this->storage);
```

Comment:
`Question:` How will this work when I am cloning a page object that is working with a content file storage and afterwards I update the cloned page? I think one would expect that still the content file gets updated, but with this, I fear it would only update the new memory storage so the changes would disappear after the request. Or am I overlooking something?

Reply:
I think this is actually avoiding an accidental overwrite of content. Let me try to explain what happens here in code:

```php
// Assuming the title of the page on disk is just "Version 1"
$a = page('some-page');

echo $page->title() // "Version 1"

// We overwrite the title on disk
$b = $a->update([
  'title' => 'Version 2'
]);

echo $a->title() // "Version 1"
echo $b->title() // "Version 2"

// At this point, $a is in memory and $b is on disk

// calling update again on $a will create another clone in memory but not make any changes on disk
$c = $a->update([
  'title' => 'Version 1.1'
]);

echo $a->title() // "Version 1" -> in memory
echo $b->title() // "Version 2" -> on disk
echo $c->title() // "Version 1.1" -> in memory

// calling update again on $b will store the changes on disk again and keep $b in memory while returning another clone $d
$d = $b->update([
  'title' => 'Version 3'
]);

echo $a->title() // "Version 1" -> in memory
echo $b->title() // "Version 2" -> in memory
echo $c->title() // "Version 1.1" -> in memory
echo $d->title() // "Version 3" -> on disk

Reply:
Your examples make sense that they'd lead to unwanted behaviour in the past. But maybe I am understanding this wrong: If we replace it on clone with a memory storage, won't $b also have the memory storage and no connection to the content file?

I think it's definitely a big breaking change that needs to be listed, if objects loose their connection to the content files once they get cloned. This might be the better/safer way forward indeed based on your examples, but I would bet a number of users runs into it where they relied upon any of the objects being able to write to the content file.

Reply:
The `::clone` method is pretty unintuitive to read, but it basically translates to this: 

```php
$this->storage = MemoryContentStorageHandler::from($this->storage);
return new static(array_replace_recursive($this->propertyData, $props));
```

This means that a completely new Page object is returned with the same properties and that new instance starts with a fresh PlainTextContentStorageHandler instance, which provides the connection to the content file. 

The main problem here really is that we introduced the idea of immutable objects in a very half-baked way. As long as cloning a Page object leads to a state where mutating the clone also mutates the origin, it's not really immutable at all. 

I would almost say that this change is fixing our terrible implementation. But of course, this could lead to breaking changes where people rely on our terrible implementation :)

Reply:
I think that we really only have two choices here: 

1) we continue with this idea of immutable objects and use the storage handlers to fix it (accepting that there will be some breaking changes)
2) move away from immutable objects entirely, which means a lot more breaking changes.

Reply:
First impulse is that I like the latter. Would work like the MemoryStorageHandler but make it clear that it's for an object that usually has a content file storage (or any non-just-in-memory storage).